Description: Do not use Term::ReadLine::Gnu on dumb terminals or emacs
Origin: https://rt.cpan.org/Public/Bug/Display.html?id=123398
Author: Hiroo HAYASHI <hayashi@cpan.org>
Author: Steve Langasek <steve.langasek@ubuntu.com>
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=166987
Bug-Debian: https://bugs.debian.org/895157
Last-Modified: 2018-04-07

--- a/Gnu.pm
+++ b/Gnu.pm
@@ -80,6 +80,11 @@
 END
     }
 }
+# use Term::ReadLine::Stub on a dumb terminal.
+# https://rt.cpan.org/Ticket/Display.html?id=123398
+BEGIN {
+    croak "dumb terminal." if(!exists $ENV{TERM} || $ENV{TERM} =~ /^(dumb|emacs|unknown)$/);
+}
 
 {
     use Exporter ();
